vnext/Shared/Networking/WinRTHttpResource.cpp

```diff
@@ -55,6 +58,138 @@ WinRTHttpResource::WinRTHttpResource(IHttpClient &&client) noexcept : m_client{s
 
 WinRTHttpResource::WinRTHttpResource() noexcept : WinRTHttpResource(winrt::Windows::Web::Http::HttpClient{}) {}
 
+#pragma region IWinRTHttpRequestFactory
+
+IAsyncOperation<HttpRequestMessage> WinRTHttpResource::CreateRequest(
+    HttpMethod &&method,
+    Uri &&uri,
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, IInspectable> props) noexcept /*override*/ {
+  auto request = HttpRequestMessage{std::move(method), std::move(uri)};
+  for (auto prop : props) {
+    request.Properties().Insert(prop.Key(), prop.Value());
+  }
+
+  auto iReqArgs = request.Properties().Lookup(L"RequestArgs");
+  auto reqArgs = iReqArgs.as<RequestArgs>();
+  auto self = shared_from_this();
+
+  HttpMediaTypeHeaderValue contentType{nullptr};
+  string contentEncoding;
+  string contentLength;
+
+  // Headers are generally case-insensitive
+  // https://www.ietf.org/rfc/rfc2616.txt section 4.2
```

Comment:
>tion 4.2 [](http://example.com/codeflow?start=45&length=8)

Should probably also comment on http/2:

[https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2](https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2)

Had to dig deep in http/1.1 spec. to find the definition for token in section 2.2
``` 
token          = 1*<any CHAR except CTLs or separators>
CHAR           = <any US-ASCII character (octets 0 - 127)>
CTL            = <any US-ASCII control character
                        (octets 0 - 31) and DEL (127)>

Reply:
only now realized that diff did not properly mark this code as 'moved'.....